src/Utilities/VariableTemplates/VariableTemplates.jl

```diff
@@ -315,11 +315,22 @@ end
     expr
 end
 
+export unroll_map
+unroll_map(f::F, N::Int, args...) where {F} = unroll_map(f, Val(N), args...)
+@generated function unroll_map(f::F, ::Val{N}, args...) where {F, N}
+    quote
+        Base.@_inline_meta
+        Base.Cartesian.@nexprs $N i -> f(Val(i), args...)
+    end
+end
+
+Base.@propagate_inbounds Base.getindex(v::AbstractVars, i::Int) =
+    Base.getindex(v, Val(i))
```

Comment:
@jkozdon I'm not sure exactly why, but this was even needed for some reason (otherwise there was a method error).

Reply:
Yeah, I dunno. I can take a crack at it next week if you want, or we can just run with the generated function. Fine by me either way.

Reply:
Yeah in the method below `i` is used for an operation on types so this makes sense.